internal/util/kubernetes/object/set.go

```diff
@@ -3,53 +3,91 @@ package object
 import "sigs.k8s.io/controller-runtime/pkg/client"
 
 type (
-	gvk       string
-	namespace string
-	name      string
+	gvk        string
+	namespace  string
+	name       string
+	generation int64
+)
+
+type object struct {
+	generation int64
+	succeeded  bool
+}
```

Comment:
Instead of storing just an object in the set, we keep its generation and a flag on whether it was successfully configured as well.